gsl/gsl_byte

```diff
@@ -106,6 +106,12 @@ constexpr IntegerType to_integer(byte b) noexcept
     return {b};
 }
 
+constexpr byte to_byte(unsigned char i) noexcept
+{
```

Comment:
I also experimented with initializer_list's to get syntax like:
```
byte d = to_byte({0x00});
```
Which I think looks better and works better. 
But allows invalid calls:
```
byte e = to_byte({});
byte f = to_byte({0x00, 0x00});
```
I can't see anyway ensure 1 entry.
@gdr-at-ms: Do options a, b or d look like they might be acceptable? Option b won't work with VS2013 :(